# AJ0006 - Probably wrong type argument for ILogger<TContext>

## Explanation

When injecting `ILogger<TContext>` into a class through the constructor, the type of `TContext` should be the type the
logger is injected to.

## Code example

````csharp
public class MyService
{
    // All members are not using 'MyService' as generic argument -> warning
    private ILogger<IAmGroot> _logger;
    private ILogger<IAmGroot> Logger { get; }
    
    public MyService(ILogger<IAmGroot> logger) 
    {
    }
}

public class MyService
{
    // All members are compliant
    private ILogger<MyService> _logger;
    private ILogger<MyService> Logger { get; }
    
    public MyService(ILogger<MyService> logger) 
    {
    }
}
````

## Configuration

| Configuration Key (.globalconfig) | Description                                          |
|:----------------------------------|:-----------------------------------------------------|
| `AJ0006.is_enabled`               | Enables or disables the analyzer. Default is `true`. |
