# AJ0006 - Probably wrong type argument for ILogger<TContext>

## Explanation

When injecting `ILogger<TContext>` into a class through the constructor, the type of `TContext` should be the type the
logger is injected to.

## Code example

````csharp
// this will yield a warning
public class MyService
{
    public MyService(ILogger<IAmGroot> logger) // Wrong generic type argument. Should be MyService
    {
    }
}

// compliant
public class MyService
{
    public MyService(ILogger<MyService> logger)
    {
    }
}
````

## Configuration

| Configuration Key (.globalconfig) | Description                                          |
|:----------------------------------|:-----------------------------------------------------|
| `AJ0006.is_enabled`               | Enables or disables the analyzer. Default is `true`. |
